### PR TITLE
Enable GitHub Actions

### PR DESCRIPTION
**Pull request by kurema**
You can use GitHub Actions to build.
Compiled APK:
https://github.com/kurema/PHEM/releases/tag/v1.43

Building for ARM64 failed. You may need some update to enable ARM64 build. ~And maybe you should sign.~

### CODE REVIEW

**Comment by kurema** [comment]
On ARM64.

>   In file included from /home/runner/work/PHEM/PHEM/app/src/main/jni/SrcShared/Hardware/EmCPU68K.cpp:14:0:
  /home/runner/work/PHEM/PHEM/app/src/main/jni/SrcShared/Hardware/EmCPU68K.cpp: In member function 'void EmCPU68K::ProcessException(ExceptionNumber)':
  /home/runner/work/PHEM/PHEM/app/src/main/jni/SrcShared/EmCommon.h:62:61: error: size of array '_dummy' is negative
   #define COMPILE_TIME_ASSERT(x) do { char _dummy[(x) ? 1 : -1]; UNUSED_PARAM (_dummy)} while (0)
                                                               ^
  /home/runner/work/PHEM/PHEM/app/src/main/jni/SrcShared/Hardware/EmCPU68K.cpp:1196:3: note: in expansion of macro 'COMPILE_TIME_ASSERT'
     COMPILE_TIME_ASSERT (sizeof (ExceptionStackFrame2) == 14);
     ^
  /home/runner/work/PHEM/PHEM/app/src/main/jni/SrcShared/EmCommon.h:62:61: error: size of array '_dummy' is negative
   #define COMPILE_TIME_ASSERT(x) do { char _dummy[(x) ? 1 : -1]; UNUSED_PARAM (_dummy)} while (0)
                                                               ^
  /home/runner/work/PHEM/PHEM/app/src/main/jni/SrcShared/Hardware/EmCPU68K.cpp:1212:3: note: in expansion of macro 'COMPILE_TIME_ASSERT'
     COMPILE_TIME_ASSERT (sizeof (ExceptionStackFrame1) == 6);
     ^

It looks like alignment issue. But `#pragma pack(2)` is written in [PalmPack.h](https://github.com/Florin9doi/PHEM/blob/master/app/src/main/jni/SrcShared/PalmPack.h), so I don't know why this happened (related [stackoverflow](https://stackoverflow.com/questions/54257525/)).

**Comment by Florin9doi** [comment]
I appreciate your interest, but the original developer has abandoned this project in [2014](https://web.archive.org/web/20191002051917/http://perpendox.com:80/phem/index.html) and this repo is mainly I backup, I have no plans to work on maintenance.

First ARM64 / x86_64 issues are caused by these defines, but there are probably many more issues, considering that this emulator was never ported to 64bit CPUs: 

- https://github.com/Florin9doi/PHEM/blob/b3f29425fa4acaa51cb62b37996e3df0ecd4e47e/app/src/main/jni/SrcShared/Palm/Platform/Incs/PalmTypes.h#L65
- https://github.com/Florin9doi/PHEM/blob/b3f29425fa4acaa51cb62b37996e3df0ecd4e47e/app/src/main/jni/SrcShared/EmTypes.h#L29

**Comment by kurema** [comment]
Thank you! I couldn't figure it out for hours. As a C# developer, I mistakenly though there's pre-defined UInit32.
But yeah, it's highly likely there's lots of other issues. Now that Google Play do not accept 32bit only APK, it wouldn't make much sense to maintain.
I just thought compiled and signed APK in release page or GitHub Actions workflow may help a few who just want it to run (like me). This app requires NDK r17c or lower (r18b abandoned gcc) and building with old NDK is bothersome.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+#Required if you use default ndk
+#    - name: fool the tool
+#      run: |
+#        cd /usr/local/lib/android/sdk/ndk-bundle/toolchains
+#        ln -s aarch64-linux-android-4.9 mips64el-linux-android
+#        ln -s arm-linux-androideabi-4.9 mipsel-linux-android
```
Comment (Florin9doi) [inline .github/workflows/ci.yml]: can you remove these lines and squash everything ?

**Comment by kurema** [comment]
By the way, you need to set following to the Secrets if you want to sign.
* KEYSTORE_BASE64
* KEYSTORE_PASSWORD
* KEY_ALIAS

Key generation (or use Android Studio):
`keytool -genkeypair -keyalg RSA -keysize 4096 -validity 36500 -alias github -keystore github.keystore -deststoretype pkcs12`

Base64:
`cat github.keystore | base64 > base64.txt`

**Comment by Florin9doi** [comment]
I set them, but build.gradle isn't reading release.keystore ... or something

**Comment by kurema** [comment]
I'm sorry. It was my fault. See https://github.com/Florin9doi/PHEM/pull/2.

**Comment by kurema** [comment]
"Re-run" refers to the same commit as the first run. So you have to press "Run workflow" again, not "Re-run jobs".

**Comment by kurema** [comment]
Secrets doesn't seems to be set any.

https://github.com/Florin9doi/PHEM/runs/3885020404
image

https://github.com/kurema/PHEM/runs/3885184929
image

**Comment by Florin9doi** [comment]
It worked. thanks

**Comment by CalculonPrime** [comment]
Not sure anyone will read this, but can the repo owner please enable Github issues to allow issues to be tracked?  Even if they won't be looked at by the repo owner, this will allow other users to communicate about specific issues related to PHEM.

For example, it looks like the Handera 330 rom (320x320) is broken on Oreo (8.0) in PHEM 1.43a, even though it worked fine on Nougot (7.1).  I'm curious whether anyone else has seen this.

**Pull request merged**